### PR TITLE
Add sharp_claw

### PR DESCRIPTION
**Pull request by Karthik99999**

### CODE REVIEW

File: data/mods/ssb/conditions.ts [hunk 1]
```diff
@@ -245,6 +245,24 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+	sharpclaw: {
+		noCopy: true,
+		onStart(pokemon) {
+			if (pokemon.species.name === 'Sneasel') {
+				this.add(`c:|${getName('sharp_claw')}|Hi, I'm Tumble! hf :D`);
+			} else {
+				this.add(`c:|${getName('sharp_claw')}|Hi, I'm Rough! gl >:)`);
+			}
+		},
+		// onSwitchOut implemented in ability instead
```
Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:257]: ik its being pedantic but please keep all of the big quotes in one place (e.g. what if irpa removes sharp_claw's ability?)

File: data/mods/ssb/scripts.ts [hunk 2]
```diff
@@ -87,7 +87,9 @@ export function changeSet(context: Battle, pokemon: Pokemon, newSet: SSBSet, cha
 	pokemon.canMegaEvo = context.actions.canMegaEvo(pokemon);
 	pokemon.canUltraBurst = context.actions.canUltraBurst(pokemon);
 	pokemon.canTerastallize = context.actions.canTerastallize(pokemon);
-	if (pokemon.name !== 'A Quag To The Past') context.add('-ability', pokemon, `${pokemon.getAbility().name}`);
+	if (!['A Quag To The Past', 'sharp_claw'].includes(pokemon.name)) {
+		context.add('-ability', pokemon, `${pokemon.getAbility().name}`);
+	}
```
Comment (HisuianZoroark) [inline data/mods/ssb/scripts.ts:92]: We should just remove this if its being a pain

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -519,6 +519,42 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+				for (const side of source.side.foeSidesWithConditions()) {
+					if (source.species.name === 'Sneasel') {
+						side.addSideCondition('spikes');
+					} else {
+						side.addSideCondition('toxicspikes');
+					}
+				}
+			},
+		},
+		secondary: {}, // allows sheer force to trigger
```
Comment (HisuianZoroark) [inline data/mods/ssb/moves.ts:553]: ?

**Review by HisuianZoroark — CHANGES_REQUESTED** [review]
Looks good! Just deal with my comments and its good to go for merge.

File: data/mods/ssb/moves.ts (hunk 3 above)
Comment (Karthik99999) [reply data/mods/ssb/moves.ts:553]: copied from ceaseless edge

Comment (HisuianZoroark) [reply data/mods/ssb/moves.ts:553]: Ah, should be fine then

File: data/mods/ssb/conditions.ts (hunk 1 above)
Comment (Karthik99999) [reply data/mods/ssb/conditions.ts:257]: I only did this so the quote would consistently appear before the set change message. I guess I could use priorities?

Comment (HisuianZoroark) [reply data/mods/ssb/conditions.ts:257]: You can just give the ability a negative priority, I think that will be fine

Comment (HisuianZoroark) [reply data/mods/ssb/conditions.ts:257]: Worst case scenario, we can either keep it as is or ask sharp if they want the forme change effect as an innate

Comment (Karthik99999) [reply data/mods/ssb/conditions.ts:257]: priority should work. another thing to note though is that it is currently unavoidable for the set change to be announced which is probably how one would expect the ability to work (like palafin). might be a low priority change for later

Comment (HisuianZoroark) [reply data/mods/ssb/conditions.ts]: yeah, sounds good

**Review by HisuianZoroark — APPROVED** [review]

**Pull request merged**